src/lib/block_time/block_time.ml

```diff
@@ -202,6 +207,10 @@ module Time = struct
     UInt64.of_int64
       (Int64.of_float (Time.Span.to_ms (Time.to_span_since_epoch t)))
 
+  (* TODO: This has two issues:
+     -) Time.t can't hold the full uint64 range, so this can fail for large t
+     -) The int conversion causes negative timestamps if the highest bit is 1
+  *)
   let to_time t =
```

Comment:
Can we make this return an `Or_error.t`, with an error if the conversion to `int64` is negative?

Reply:
Making it an `Orr_error.t` would have required me to change restructure the code in ~40-50 places where it is used, which is code that I'm not familiar with. I've opted for leaving function signatures the same and just changing the name to `to_time_exn`, so it's apparent now that this throws an error. I'd like to leave the refactor to `Or_error.t` to the protocol team / another PR if that's OK